tests/functional/lib/helpers.js

```diff
@@ -696,45 +696,46 @@ define([
     var age = options.age || 24;
     var submit = options.submit !== false;
 
-    var remote = getRemote(context);
-    return remote
-      .getCurrentUrl()
-      .then(function (currentUrl) {
-        // only load the signup page if not already at a signup page.
-        // the leading [\/#] allows for either the standard redirect or iframe
-        // flow. The iframe flow must use the window hash for routing.
-        if (! /[\/#]signup(?:$|\?)/.test(currentUrl)) {
-          return remote
-            .get(require.toUrl(SIGNUP_URL))
-            .setFindTimeout(intern.config.pageLoadTimeout);
-        }
-      })
+    return function () {
+      return this.parent
+        .getCurrentUrl()
+        .then(function (currentUrl) {
+          // only load the signup page if not already at a signup page.
+          // the leading [\/#] allows for either the standard redirect or iframe
+          // flow. The iframe flow must use the window hash for routing.
+          if (! /[\/#]signup(?:$|\?)/.test(currentUrl)) {
+            return this.parent
```

Comment:
Yup, this is in fact totally legit, which is why I'm able to remove the `context` parameter. `then`'s are called with a context that contains a `parent` property that references what the `context` parameter used to. You can see other examples of `this.parent` inside of a then function sprinkled throughout helpers.js